include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -893,4 +928,115 @@ operator()(
                 });
     }
 }
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSplines,
+        class InterpolationDDim,
+        ddc::BoundCond BcLower,
+        ddc::BoundCond BcUpper,
+        SplineSolver Solver,
+        class... IDimX>
+std::tuple<
+        ddc::Chunk<
+                double,
+                ddc::DiscreteDomain<
+                        ddc::Deriv<typename InterpolationDDim::continuous_dimension_type>>,
+                ddc::KokkosAllocator<double, MemorySpace>>,
+        ddc::Chunk<
+                double,
+                ddc::DiscreteDomain<InterpolationDDim>,
+                ddc::KokkosAllocator<double, MemorySpace>>,
+        ddc::Chunk<
+                double,
+                ddc::DiscreteDomain<
+                        ddc::Deriv<typename InterpolationDDim::continuous_dimension_type>>,
+                ddc::KokkosAllocator<double, MemorySpace>>>
+SplineBuilder<
+        ExecSpace,
+        MemorySpace,
+        BSplines,
+        InterpolationDDim,
+        BcLower,
+        BcUpper,
+        Solver,
+        IDimX...>::quadrature_coefficients() const
+{
+    // Compute integrals of bsplines
+    ddc::Chunk<double, ddc::DiscreteDomain<bsplines_type>> integral_bsplines(spline_domain());
+    ddc::discrete_space<bsplines_type>().integrals(integral_bsplines.span_view());
+
+    // Remove last cell in the periodic case
+    ddc::ChunkSpan integral_bsplines_without_periodic_point
+            = integral_bsplines.span_view()[ddc::DiscreteDomain<bsplines_type>(
+                    ddc::DiscreteElement<bsplines_type>(0),
+                    ddc::DiscreteVector<bsplines_type>(matrix->size()))];
+
+    // Allocate mirror with additional rows (cf. SplinesLinearProblem3x3Blocks documentation)
+    Kokkos::View<double**, Kokkos::LayoutRight, Kokkos::DefaultHostExecutionSpace>
+            integral_bsplines_mirror_with_additional_allocation(
+                    "integral_bsplines_mirror_with_additional_allocation",
+                    matrix->required_number_of_rhs_rows(),
+                    1);
+
+    // Extract relevant subview
+    Kokkos::View<double*, Kokkos::LayoutRight, Kokkos::DefaultHostExecutionSpace>
+            integral_bsplines_mirror = Kokkos::
+                    subview(integral_bsplines_mirror_with_additional_allocation,
+                            std::pair<std::size_t, std::size_t> {
+                                    0,
+                                    integral_bsplines_without_periodic_point.size()},
+                            0);
+
+    // Solve matrix equation A^t*X=integral_bsplines
+    Kokkos::deep_copy(
+            integral_bsplines_mirror,
+            integral_bsplines_without_periodic_point.allocation_kokkos_view());
+    matrix->solve(integral_bsplines_mirror_with_additional_allocation, true);
```

Comment:
If the matrix is stored on GPU won't it be a problem that integral_bsplines_mirror_with_additional_allocation is on CPU?

Reply:
You are correct, I was not attentive when doing that, thx

Reply:
Now it performs the computation on the ExecSpace but take an optional template argument OutMemorySpace, such that the computation may be performed and CPU but the coefficients are returned on GPU.

Reply:
This looks better :+1: